tiledb/sm/query/query_buffer.h

```diff
@@ -147,6 +147,32 @@ class QueryBuffer {
     return *this;
   }
 
+  /** Move-assignment Operator. */
+  QueryBuffer& operator+=(QueryBuffer&& rhs) {
```

Comment:
Do not use an `operator +=` because it can not return a `Status`. We need to handle the scenario where the `rhs` contains a buffer that is already set on `this`.

Additionally, this incurs a small amount of unnecessary overhead in the current usage. If we have an existing `QueryObject` and want to attach a new buffer and size to it, we are unnecessarily constructing a second object to add it to, For example:
```
buffers_[name] += QueryBuffer(buffer, nullptr, buffer_size, nullptr);
```
should be something like
```
buffers_[name].set_buffer_data(buffer, buffer_size);
```